### ISSUE TITLE
Training Script

### ISSUE THREAD

**Issue by ganyu-gs** [PeiwenSun2000/Both-Ears-Wide-Open#3]
Hi，很高兴能看到这样的工作，还获得了spotlight，但是不知道是否有具体的训练代码呢，如果能告诉我，将不甚感激！

**Comment by PeiwenSun2000** [comment]
Since the stable audio code is still being improved, I would say the instructions for training in [stable-audio-tools](https://github.com/Stability-AI/stable-audio-tools) are detailed.

And beyond that. I would like to provide some snippets of my training command.

```
export CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
python3 ./train.py \
    --num-gpus 8 \
    --batch-size 64 \
    --dataset-config data_cofig_template.json \
    --model-config model_cofig_template.json \
    --name name_here \
    --save-dir dir_here \
    --strategy deepspeed \
    --pretrained-ckpt-path ~/stable_audio_open/model.ckpt 
    # model.ckpt from https://huggingface.co/stabilityai/stable-audio-open-1.0
    # or torch lightning ckpt to continue training
    # --ckpt-path ../2mr7i5l2/checkpoints/epoch=83-step=440000.ckpt
```

**Comment by ganyu-gs** [comment]
In this paper, you mentioned Azimuth Information: encoder, I wanne know how to train it. And finetune the pretrain model.

**Comment by PeiwenSun2000** [comment]
The training of azimuth information is related to two perspectives.

1. Data: Feel free to download the [dataset](https://github.com/PeiwenSun2000/Both-Ears-Wide-Open/tree/main/datasets). The metadata includes azimuth labels.
2. Model: The only part related to azimuth is [conditioners.py](https://github.com/PeiwenSun2000/Both-Ears-Wide-Open/blob/main/models/stable_audio_tools/models/conditioners.py)

I hope this helps.

**Comment by BuJiaLaDiii** [comment]
hi there , I tried the data in `full_move_filtered.jsonl` , But when  I try the PostionConditioner there some error. the input and the data I used as below.


```
    pos = PostionConditioner(output_dim=768)
    cond_text = torch.randn(1, 128, 768)
    floats = [
        [
            (65.3076, 41.7769, 1.0, 1),  
            (49.6091, 51.7608, 1.0, 1), 
            (38.7347, 38.7347, 0, 1),  
        ]
    ]
    out = pos( cond_text, floats,'cpu')
```
```
{"audio_name":"45a192066544305803bd89981c6da6f3","meta":"BBC&07043127","
caption":"A car passes by from right to left at a moderate speed.","size":null,
"room_size":[91.5376643552,93.7007409466,94.6674375414],"micro_pos":[[47.3052807638,47.4663673712],[51.6359658128,51.6359658128],[38.7347006691,38.7347006691]],

"start":[65.3076257448,49.6090774687,38.7347006691,0.4287685875],
"end":[41.7769351226,51.760761102,38.7347006691,0.1333723305],

"RT60":0.5378422567,"change_step":[0],
"raw_start":[5],
"raw_end":[1],
"start_angle":[-6.4525194431],
"end_angle":[178.7254046388],
"raw_speed":["moderate"],
"speed_ratio":[0.5269798935],
"move_start_time":[23738],
"raw_audio_path":["\/data\/peiwensun\/dataset\/processed_data\/BBC\/wav\/07043127.wav"]}

```

And I also tried another input, 
```
    floats = [
        [
            (1.0, 16.0, 1.0, 1),  
            (5.0, 10.0, 1.0, 0),  
            (8.0, 8.0, 0, 1),  
            (3.0, 15.0,0, 1)  
        ]
    ]
```
I wonder is there a method to mapping the physical position, like `"start":[65.3076257448,49.6090774687,38.7347006691,0.4287685875]`, into the correct input.

**Comment by PeiwenSun2000** [comment]
In our training, our azimuth input is set to [1, 5] int (for coarse) and [1, 5] float (for fine). You can observe the input by providing inference examples.

**Processed projection:** Use the `raw_start` and `raw_end` (for coarse); use the `start_angle` and `end_angle` (for fine). The definitions of each column name can be found on [Hugging Face](https://huggingface.co/datasets/spw2000/BEWO-1M).

**Original information:** The accurate location and other specific details like `room_size`,  `start` and `end`are only for potential further research.

Feel free to use the processed projection and original information in any way you wish.
